File: FoodForecastingAPI/FoodForecastingAPI/Models/Redistribution.cs
namespace FoodForecastingAPI.Models
{
    public class Redistribution
    {
        public int Id { get; set; }

        public DateTime RedistributionDate { get; set; }

        public string MealName { get; set; } = "";

        public int Quantity { get; set; }

        public string Destination { get; set; } = "";

        public string ReceiverName { get; set; } = "";

        public string Status { get; set; } = "Distributed";
    }
}

File: food-forecasting-ui/src/app/services/redistribution.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Redistribution {
  id?: number;
  redistributionDate: string;
  mealName: string;
  quantity: number;
  destination: string;
  receiverName: string;
  status: string;
}

@Injectable({
  providedIn: 'root'
})
export class RedistributionService {

  private apiUrl =
    'http://localhost:5003/api/Redistributions';

  constructor(private http: HttpClient) {}

  getRedistributions(): Observable<Redistribution[]> {
    return this.http.get<Redistribution[]>(this.apiUrl);
  }

  addRedistribution(
    redistribution: Redistribution
  ): Observable<Redistribution> {
    return this.http.post<Redistribution>(
      this.apiUrl,
      redistribution
    );
  }

  deleteRedistribution(id: number): Observable<void> {
    return this.http.delete<void>(
      `${this.apiUrl}/${id}`
    );
  }
}